### PR TITLE
Feat/api-client

### PR DESCRIPTION
**Pull request by henriq4**

### CODE REVIEW

File: apps/mobile/lib/modules/core/core_module.dart [hunk 1]
```diff
@@ -2,6 +2,8 @@ import 'package:flutter_modular/flutter_modular.dart';
 import 'package:itourism_mobile/infra/notification/notification_service.dart';
 import 'package:shared_preferences/shared_preferences.dart';
 
+import '../../infra/api/http/api_client.dart';
```
Comment (eduardoteles17) [inline apps/mobile/lib/modules/core/core_module.dart:5]: Prefer absolute imports

File: apps/mobile/lib/modules/core/core_module.dart [hunk 2]
```diff
@@ -10,5 +12,6 @@ class CoreModule extends Module {
           export: true,
         ),
         Bind.lazySingleton((i) => NotificationService(), export: true),
+        Bind.lazySingleton((i) => ApiClient(), export: true),
```
Comment (eduardoteles17) [inline apps/mobile/lib/modules/core/core_module.dart:15]: Use the static  method to instance the api client

File: apps/mobile/lib/modules/core/core_module.dart (hunk 1 above)
Comment (eduardoteles17) [inline apps/mobile/lib/modules/core/core_module.dart:5]: Prefer absolute imports

**Review by eduardoteles17 — CHANGES_REQUESTED** [review]
Need small changes

File: apps/mobile/lib/modules/core/core_module.dart (hunk 2 above)
Comment (henriq4) [reply apps/mobile/lib/modules/core/core_module.dart]: Resolved.

File: apps/mobile/lib/modules/core/core_module.dart (hunk 1 above)
Comment (henriq4) [reply apps/mobile/lib/modules/core/core_module.dart]: Resolved.

**Review by eduardoteles17 — APPROVED** [review]
Ok

**Pull request merged**